nixos/modules/services/misc/prowlarr.nix

```
@@ -31,7 +37,7 @@ in
         Type = "simple";
         DynamicUser = true;
         StateDirectory = "prowlarr";
```

Systemd probaly doesn't need to create a state directory if the data dir is not the default.

```suggestion
        StateDirectory = lib.mkIf (cfg.dataDir == "/var/lib/prowlarr"") "prowlarr";
```